src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -395,11 +235,48 @@ public virtual bool ShouldConvertChallengeToForbidden()
         /// deals an authentication interaction as part of it's request flow. (like adding a response header, or
         /// changing the 401 result to 302 of a login page or external sign-in location.)
         /// </summary>
-        /// <returns></returns>
-        protected virtual Task ApplyResponseChallengeAsync()
+        /// <param name="context"></param>
+        /// <returns>True if no other handlers should be called</returns>
+        protected virtual Task<bool> HandleUnauthorizedAsync(ChallengeContext context)
         {
-            ApplyResponseChallenge();
-            return Task.FromResult(0);
+            Response.StatusCode = 401;
```

Comment:
Well today it would be you get a 403 as the status code?  I'm ok with overidding the default behavior to just throw NotSupported/InvalidOperation instead on the OAuth providers if we think that's better?